main.go

```diff
@@ -1,235 +1,237 @@
-	pause = true
-	log.Println("Pausing from web request")
-	w.Write([]byte("Ok, paused.\n"))
-}
-
-func ResumeHandler(w http.ResponseWriter, r *http.Request) {
-	pause = false
-	log.Println("Unpausing from web request")
-	w.Write([]byte("Ok, unpaused.\n"))
-}
-func SkipHandler(w http.ResponseWriter, r *http.Request) {
-	log.Println("Skippingfrom web request")
-	pause = false
-	abort <- struct{}{}
-
-	w.Write([]byte("Skipping current programming and resume program list runner from web request.\n"))
-}
-
-// Control channel to stop running programs immediately (yes, global)
-var abort = make(chan struct{})
-var pause bool = false
-
-func main() {
-
-	InitializeConfig()
-
-	go LoadAndRunLoop()
-
-	r := mux.NewRouter()
-	r.HandleFunc("/play", PlayHandler)
-	r.HandleFunc("/pause", PauseHandler)
-	r.HandleFunc("/resume", ResumeHandler)
-	r.HandleFunc("/skip", SkipHandler)
-
-	go log.Fatal(http.ListenAndServe(":8080", r))
+// package main
+
+// import (
+// 	"bufio"
+// 	"encoding/csv"
+// 	"fmt"
+// 	"io"
+// 	"io/ioutil"
+// 	"log"
+// 	"net"
+// 	"net/http"
+// 	"os"
+// 	"strings"
+// 	"time"
+
+// 	"github.com/fsnotify/fsnotify"
+// 	"github.com/gorilla/mux"
+// 	"github.com/spf13/viper"
+// )
+
+// type Program struct {
+// 	URL      string
+// 	Duration time.Duration
+// }
+
+// // InitializeConfig loads our configuration using Viper package.
+// func InitializeConfig() {
+
+// 	viper.SetConfigType("yaml")
+// 	// Set config file
+// 	viper.SetConfigName("config")
+
+// 	// Add config path
+// 	//	viper.AddConfigPath("$HOME/.gorotator")
+// 	viper.AddConfigPath(".")
+
+// 	// Read in the config
+// 	err := viper.ReadInConfig() // Find and read the config file
+// 	if err != nil {             // Handle errors reading the config file
+// 		panic(fmt.Errorf("Fatal error config file: %s \n", err))
+// 	}
+
+// 	// Load default settings
+// 	viper.SetDefault("debug", false)
+
+// 	viper.SetEnvPrefix("gorotator") // will be uppercased automatically
+// 	viper.BindEnv("debug")
+// 	viper.BindEnv("ip")
+// 	viper.BindEnv("port")
+
+// 	// Do some flag handling and any complicated config logic
+// 	if !viper.IsSet("ip") || !viper.IsSet("port") {
+// 		fmt.Println("Configuration error.  Both IP and PORT must be set via either config or environment.")
+// 		os.Exit(1)
+// 	}
+
+// 	viper.WatchConfig()
+// 	viper.OnConfigChange(func(e fsnotify.Event) {
+// 		fmt.Println("\nConfig file changed:", e.Name)
+// 		abort <- struct{}{}
+// 		fmt.Printf("Content will change immediately.\n\n")
+// 	})
+
+// }
+
+// // Loads a list of programs.
+// // A program consists of a list things to display on the rotator along
+// // with a number of seconds to display each one before moving on.
+// func loadProgramList(filename string) []Program {
+
+// 	fmt.Printf("Loading programs from %s\n", filename)
+
+// 	var list []Program
+
+// 	bytes, err := ioutil.ReadFile(filename)
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+// 	webpages := string(bytes)
```

Comment:
`bytes.NewReader(bytes)` below obviates the need for this. (You'll need to rename the variable of course, once you're importing `bytes`.)